library/core/src/num/int_macros.rs

```diff
@@ -2269,6 +2269,45 @@ macro_rules! int_impl {
             }
         }
 
+        /// Calculates the remainder of `self / rhs` if the quotient is rounded toward negative infinity.
+        ///
+        /// # Panics
+        ///
+        /// This function will panic if `rhs` is zero.
+        ///
+        /// ## Overflow behavior
+        ///
+        /// On overflow, this function will panic if overflow checks are enabled (default in debug
```

Comment:
Signed `%` always panics on overflow (MIN / -1) regardless of the overflow checks compile option. At least this particular method will panic iff the `%` inside it panics. Therefor, the documentation is incorrect. The already merged ceil and floor div functions should be double checked with this in mind.

Once that is fixed I suggest explaining exactly what constitutes an overflow. (The remainder function never *truly* overflows, but it panics whenever the corresponding division would also panic.)

Reply:
> The already merged ceil and floor div functions should be double checked with this in mind.

Submitted a PR for that and some other stuff. #119726